### PR TITLE
First four tree samples

### PR DESCRIPTION
**Pull request by tomdicarlo**

### CODE REVIEW

File: src/frontend-samples/tree-samples/custom-table-node-tree/sampleSpec.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export function getCustomTableNodeTreeSpec(): SampleSpec {
+  return ({
+    name: "custom-table-node-tree-sample",
+    label: "Custom Table Node Tree",
+    image: "custom-table-node-tree-thumbnail.png",
+    customModelList: [],
+    files: [
+      { name: "CustomTableNodeTreeSample.tsx", import: import("!!raw-loader!./index") },
+      { name: "Common.tsx", import: import("!!raw-loader!../Common") },
```
Comment (Josh-Schifter) [inline src/frontend-samples/tree-samples/custom-table-node-tree/sampleSpec.tsx:12]: This one doesn't use common, right?

Comment (tomdicarlo) [reply src/frontend-samples/tree-samples/custom-table-node-tree/sampleSpec.tsx]: You're right, just fixed that

File: src/frontend-samples/tree-samples/basic-tree/sampleSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return ({
+    name: "basic-tree-sample",
+    label: "Basic Tree",
+    image: "basic-tree-thumbnail.png",
+    customModelList: [],
+    files: [
+      { name: "BasicTreeSample.tsx", import: import("!!raw-loader!./index") },
+      { name: "Common.tsx", import: import("!!raw-loader!../Common") },
+      { name: "index.scss", import: import("!!raw-loader!../index.scss") },
+      { name: "GithubLink.tsx", import: import("!!raw-loader!../../../Components/GithubLink") },
```
Comment (Josh-Schifter) [inline src/frontend-samples/tree-samples/basic-tree/sampleSpec.tsx:15]: Please remove GithubLink.tsx

File: src/frontend-samples/tree-samples/custom-event-handler-tree/sampleSpec.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  return ({
+    name: "custom-event-handler-tree-sample",
+    label: "Custom Event Handler Tree",
+    image: "custom-event-handler-tree-thumbnail.png",
+    customModelList: [],
+    files: [
+      { name: "CustomEventHandlerTreeSample.tsx", import: import("!!raw-loader!./index") },
+      { name: "Common.tsx", import: import("!!raw-loader!../Common") },
+      { name: "index.scss", import: import("!!raw-loader!../index.scss") },
+      { name: "GithubLink.tsx", import: import("!!raw-loader!../../../Components/GithubLink") },
```
Comment (Josh-Schifter) [inline src/frontend-samples/tree-samples/custom-event-handler-tree/sampleSpec.tsx:14]: Please remove GithubLink.tsx

File: src/frontend-samples/tree-samples/custom-table-node-tree/sampleSpec.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  return ({
+    name: "custom-table-node-tree-sample",
+    label: "Custom Table Node Tree",
+    image: "custom-table-node-tree-thumbnail.png",
+    customModelList: [],
+    files: [
+      { name: "CustomTableNodeTree.tsx", import: import("!!raw-loader!./index") },
+      { name: "index.scss", import: import("!!raw-loader!../index.scss") },
+      { name: "TableNodeTree.scss", import: import("!!raw-loader!./TableNodeTree.scss") },
+      { name: "GithubLink.tsx", import: import("!!raw-loader!../../../Components/GithubLink") },
```
Comment (Josh-Schifter) [inline src/frontend-samples/tree-samples/custom-table-node-tree/sampleSpec.tsx:14]: Please remove GithubLink.tsx

File: src/frontend-samples/tree-samples/custom-checkboxes-tree/sampleSpec.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return ({
+    name: "custom-checkboxes-sample",
+    label: "Custom Checkboxes Tree",
+    image: "custom-checkboxes-tree-thumbnail.png",
+    customModelList: [],
+    files: [
+      { name: "CustomCheckboxesTreeSample.tsx", import: import("!!raw-loader!./index") },
+      { name: "Common.tsx", import: import("!!raw-loader!../Common") },
+      { name: "index.scss", import: import("!!raw-loader!../index.scss") },
+      { name: "GithubLink.tsx", import: import("!!raw-loader!../../../Components/GithubLink") },
```
Comment (Josh-Schifter) [inline src/frontend-samples/tree-samples/custom-checkboxes-tree/sampleSpec.tsx:14]: Please remove GithubLink.tsx

File: src/frontend-samples/tree-samples/custom-event-handler-tree/sampleSpec.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { SampleSpec } from "../../../Components/SampleShowcase/SampleShowcase";
```
Comment (calebmshafer) [inline src/frontend-samples/tree-samples/custom-event-handler-tree/sampleSpec.tsx:1]: Missing copyright header.

File: src/frontend-samples/tree-samples/custom-checkboxes-tree/sampleSpec.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { SampleSpec } from "../../../Components/SampleShowcase/SampleShowcase";
```
Comment (calebmshafer) [inline src/frontend-samples/tree-samples/custom-checkboxes-tree/sampleSpec.tsx:1]: Missing copyright header.  I need to fix the build check for this...

**Pull request merged**